libretroshare/src/jsonapi/jsonapi.cpp

```diff
@@ -84,13 +92,13 @@ static bool checkRsServicePtrReady(
 	return false;
 }
 
-
-JsonApiServer::JsonApiServer(
-        uint16_t port, const std::string& bindAddress,
-        const std::function<void(int)> shutdownCallback ) :
-    mPort(port), mBindAddress(bindAddress), mShutdownCallback(shutdownCallback)
+JsonApiServer::JsonApiServer(uint16_t port, const std::string& bindAddress,
+        const std::function<bool(const std::string&)> newAccessRequestCallback ) :
+    mPort(port), mBindAddress(bindAddress),
+    mNewAccessRequestCallback(newAccessRequestCallback),
+    configMutex("JsonApiServer config")
 {
```

Comment:
you forgot to update `WebuiPage.cpp`
```
../../../RetroShare/retroshare-gui/src/gui/settings/WebuiPage.cpp: In static member function 'static bool WebuiPage::checkStartWebui()':
../../../RetroShare/retroshare-gui/src/gui/settings/WebuiPage.cpp:121:54: error: no matching function for call to 'JsonApiServer::JsonApiServer(int, const char*, WebuiPage::checkStartWebui()::<lambda(int)>)'
              [](int /*ec*/) { std::raise(SIGTERM); } );
                                                      ^
```

Reply:
not used there anymore ;) now JSON API has it's own configuration page